lib/audit_test.pm

```diff
@@ -47,78 +48,102 @@ sub run_testcase {
     my ($testcase, %args) = @_;
 
     # Run test case
-    assert_script_run("cd ${testdir}${testfile_tar}/audit-test/${testcase}/");
-    assert_script_run('./run.bash', %args);
+    assert_script_run("cd ${testdir}${testfile_tar}/audit-test/");
+    assert_script_run('make') if ($args{make});
+    assert_script_run("cd ${testcase}/");
+
+    # Export MODE
+    assert_script_run("export MODE=$audit_test::mode");
+
+    assert_script_run('./run.bash', timeout => $args{timeout});
 
     # Upload logs
     upload_logs("$current_file");
     upload_logs("$baseline_file");
 }
 
-# Parse all test cases in *.log (run.log, etc)
-# input: $file - file name
-# output: @testcase_list - test case list
-sub parse_testcase_log {
-    my ($file) = @_;
-
-    # Test case token, it is a number within []: e.g., [0], [1], [12]
-    my $s_tok         = '\[\d+\] ';
-    my @testcase_list = ();
-
-    my @lines = split(/\n/, path("$file")->slurp);
-    my $i     = 0;
-    # Parse the test cases of input file
-    for my $line (@lines) {
-        if ($line =~ /$s_tok/) {
-            push @testcase_list, "$line\n";
-            $i++;
+sub parse_lines {
+    my ($lines) = @_;
+    my @results;
+    foreach my $line (@$lines) {
+        if ($line =~ /(\[\d+\])\s+(.*)\s+(PASS|FAIL|ERROR)/) {
+            my $name = $2;
+            $name = trim $name;
+            push @results, {id => $1, name => $name, result => $3};
```

Comment:
```suggestion
            push @results, {id => $1, name => trim $2, result => $3};
```

should be enough

Reply:
I tried this on my local. If I put `trim $2` inside the hash directly `{id => $1, name => trim $2, result => $3}`  the `result` will be missed. such as:
```
$VAR1 = [
          {
            'id' => '[0]',
            'name' => 'auditd_restart'
          },
          {
            'id' => '[1]',
            'name' => 'auditd_start'
          },
    ]
```
So I put that outside the hash.  Does it make sense?

Reply:
ok, the "trim" function takes the next parameters unless you surround the correct argument with parentheses

```
push @results, {id => $1, name => trim($2), result => $3};
```